ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vrfs.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.network_services.utils import UtilsMixin
+
+
+class VrfsMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def vrfs(self):
+        """
+        Return structured config for vrfs
+
+        Used for creating VRFs except VRF "default"
+        """
+
+        if not self._network_services_l3:
+            return None
+
+        vrfs = {}
+        for tenant in self._filtered_tenants:
+            for vrf in tenant["vrfs"]:
+                vrf_name = vrf["name"]
+                if vrf_name == "default":
+                    continue
+
+                new_vrf = {
+                    "tenant": tenant["name"],
+                    "ip_routing": True,
+                }
+                if vrf_name in self._ipv6_vrf_names:
+                    new_vrf["ipv6_routing"] = True
+
+                if "description" in vrf:
+                    new_vrf["description"] = vrf["description"]
+
+                vrfs[vrf_name] = new_vrf
+
+        if vrfs:
+            return vrfs
+
+        return None
+
+    @cached_property
+    def _ipv6_vrf_names(self) -> set:
```

Comment:
wondering if having a method _has_ipv6 to be called line 36 on the vrf object would not be more efficient than this cached_property (except if it is used somewhere else?)

Reply:
fixed